src/main/java/io/cdap/plugin/gcp/bigquery/actions/BigQueryCreateDataset.java

```diff
@@ -0,0 +1,185 @@
+import com.google.cloud.bigquery.DatasetInfo;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.etl.api.action.Action;
+import io.cdap.cdap.etl.api.action.ActionContext;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryUtil;
+import io.cdap.plugin.gcp.common.GCPConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * This class <code>BigQueryCreateDataset</code> drops a BigQuery dataset.
+ * <p>
+ *  <code>BigQueryCreateDataset</code> can be used to drop BQ dataset in other projects.
+ * </p>
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name(BigQueryCreateDataset.NAME)
+@Description(BigQueryCreateDataset.DESC)
+public class BigQueryCreateDataset extends Action {
+  private static final Logger LOG = LoggerFactory.getLogger(BigQueryCreateDataset.class);
+  public static final String NAME = "BigQueryCreateDataset";
+  public static final String DESC = "Create a Cloud BigQuery Dataset";
+  private static final String RECORDS_OUT = "records.out";
+  private Config config;
+
+  /**
+   * Creates a dataset.
+   *
+   * @param context a <code>ActionContext</code> instance during runtime.
+   * @throws Exception thrown when delete of dataset fails.
+   */
+  @Override
+  public void run(ActionContext context) throws Exception {
+    BigQuery bq = BigQueryUtil.getBigQuery(config.getServiceAccountFilePath(), config.getProject());
+    try {
+      Dataset dataset = bq.getDataset(config.getDatasetId());
+      if (dataset == null) {
+        dataset = bq.create(config.getDatasetInfo());
+        try {
+          Dataset.Builder builder = dataset.toBuilder();
+          if (config.getDescription() != null) {
+            builder.setDescription(config.getDescription());
+          }
+          if (config.getLabels().size() > 0) {
+            builder.setLabels(config.getLabels());
+          }
+          // update dataset.
+          Dataset updateDataset = builder.build();
+          updateDataset.update();
+        } catch (BigQueryException e) {
+          // we don't fail if we cannot update dataset.
+          LOG.warn(
+            String.format("Dataset '%s' was created, but failed to update metadata for dataset.",
+                          config.getDatasetName())
+          );
+        }
+        context.getMetrics().count(RECORDS_OUT, 1);
+      } else {
+        if (config.failIfExists()) {
+          throw new Exception(
+            String.format(
+              "Dataset '%s' already exists. Failing pipeline as per user request.", config.getDatasetName()
+            )
+          );
+        }
+      }
+    } catch (BigQueryException e) {
+      throw new Exception(
+        String.format(
+          "Failed to create dataset '%s'. Reason : %s", config.getDatasetName(), e.getMessage()
+        )
```

Comment:
the exception should always be included as the cause, otherwise the root cause is lost.